packages/yourloops/lib/patient/patient.util.ts

```diff
@@ -102,38 +106,49 @@ export default class PatientUtils {
+    }
+    if (patientFilters.timeOutOfTargetEnabled && patientFilters.dataNotTransferredEnabled) {
+      return patients.filter(patient => patient.alarms?.timeSpentAwayFromTargetActive || patient.alarms?.nonDataTransmissionActive)
+    }
+    if (patientFilters.timeOutOfTargetEnabled) {
+      return patients.filter(patient => patient.alarms?.timeSpentAwayFromTargetActive)
+    }
+    if (patientFilters.hypoglycemiaEnabled) {
+      return patients.filter(patient => patient.alarms?.frequencyOfSevereHypoglycemiaActive)
+    }
+    return patients.filter(patient => patient.alarms?.nonDataTransmissionActive)
   }
 
-  static extractPatients = (patients: Patient[], filterType: PatientListFilters, flaggedPatients: string[], selectedTeamId?: string): Patient[] => {
-    const twoWeeksFromNow = new Date()
-    switch (filterType) {
-      case PatientListFilters.All:
-        return selectedTeamId ? PatientUtils.getNonPendingPatients(patients, selectedTeamId) : patients
-      case PatientListFilters.Pending:
-        return PatientUtils.getPendingPatients(patients, selectedTeamId)
-      case PatientListFilters.Flagged:
-        return patients.filter(patient => flaggedPatients.includes(patient.userid))
-      case PatientListFilters.UnreadMessages:
-        return patients.filter(patient => patient.metadata.hasSentUnreadMessages)
-      case PatientListFilters.OutOfRange:
-        return patients.filter(patient => patient.alarms.timeSpentAwayFromTargetActive)
-      case PatientListFilters.SevereHypoglycemia:
-        return patients.filter(patient => patient.alarms.frequencyOfSevereHypoglycemiaActive)
-      case PatientListFilters.DataNotTransferred:
-        return patients.filter(patient => patient.alarms.nonDataTransmissionActive)
-      case PatientListFilters.RemoteMonitored:
-        return patients.filter(patient => patient.monitoring?.enabled)
-      case PatientListFilters.Private:
-        return patients.filter(patient => PatientUtils.isInTeam(patient, filterType))
-      case PatientListFilters.Renew:
-        twoWeeksFromNow.setDate(twoWeeksFromNow.getDate() + 14)
-        return patients.filter(patient => patient.monitoring?.enabled && patient.monitoring.monitoringEnd && new Date(patient.monitoring.monitoringEnd).getTime() - twoWeeksFromNow.getTime() < 0)
-      default:
-        return patients
+  static extractPatients = (patients: Patient[], patientFilters: PatientsFilters, flaggedPatientsId: string[] | undefined, selectedTeamId: string): Patient[] => {
+    // When the filter is pending, we only get the pending patients and don't apply any filter on them
+    if (patientFilters.pendingEnabled) {
+      return patients.filter((patient) => PatientUtils.isInvitationPending(patient, selectedTeamId))
+    }
+    // We do not take the pending patients
+    let patientsExtracted = PatientUtils.getNonPendingPatients(patients, selectedTeamId)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏I found another way of doing the conditional filtering that seems shorter and avoids the reassignment via `let`, tell me what you think!

```ts
const nonPendingPatients = PatientUtils.getNonPendingPatients(patients, selectedTeamId)
    
return PatientUtils.filterPatientsOnMonitoringAlerts(nonPendingPatients, patientFilters)
  .filter(patient => patientFilters.telemonitoredEnabled ? patient.monitoring?.enabled : patient)
  .filter(patient => patientFilters.manualFlagEnabled ? flaggedPatientsId?.includes(patient.userid) : patient)
  .filter(patient => patientFilters.messagesEnabled ? patient.metadata.hasSentUnreadMessages : patient)
```